src/liger_kernel/ops/cross_entropy.py

```diff
@@ -254,6 +279,23 @@ def cross_entropy_forward(
         z_loss_1d = loss_1d  # dummy ptr when return_z_loss == False
 
     n_non_ignore = (target != ignore_index).sum().item()
+    sum_of_non_ignore_weight = n_non_ignore
+    if weight is not None:
+        assert (
+            weight.shape[0] == V
+        ), f"If given, weight has to be a Tensor of size V. Got: {weight.shape}"
+        assert torch.is_floating_point(
+            weight
+        ), f"If given, weight has to be a Tensor of floating point dtype. Got: {weight.dtype}"
+        if ignore_index >= 0 and ignore_index < V:
+            weight_mask = torch.ones_like(weight)
+            weight_mask[ignore_index] = 0
+            selected_weight = torch.gather(weight * weight_mask, dim=-1, index=target)
```

Comment:
Wondering if there's any benefit to doing a torch.gather here compared to summing over `weight` after setting:

`weight[ignore_index]` = 0.

Reply:
Hmmm, but we don't compute the gradient on the weight tensor, so I'm guessing that's ok.

Reply:
This part of code needs to be fixed. Becuase if there's a out of range index in `target` like `-100`, it will omit error.